be-java-springboot/Jenkinsfile.template

```diff
@@ -26,7 +26,7 @@ def stageBuild(def context) {
   }
   stage('Build and Unit Test') {
     withEnv(["TAGVERSION=${context.tagversion}", "NEXUS_HOST=${context.nexusHost}", "NEXUS_USERNAME=${context.nexusUsername}", "NEXUS_PASSWORD=${context.nexusPassword}", "JAVA_OPTS=${javaOpts}","GRADLE_TEST_OPTS=${gradleTestOpts}","ENVIRONMENT=${springBootEnv}"]) {
-      def status = sh(script: "./gradlew clean build --stacktrace --no-daemon", returnStatus: true)
+      def status = sh(script: "use-j17.sh && ./gradlew clean build --stacktrace --no-daemon", returnStatus: true)
```

Comment:
Interesting idea... I mean you could even check the jar artifact for the target jdk it was built for and decide depending on that automatically, but for me these solutions are a bit hacky and also make traceability more complex.

Reply:
@renedupont I can't follow. The jar artifact does not exists at this point. It will be created by calling gradle. Can you please elaborate a bit more?

btw... have a quick look at the script `use-j17.sh`. It is just 4 lines. It is effective and short and same instructions used in dockerfile to set the java  version.

Reply:
Oh yes, somehow I was only thinking about the jdk version of the docker container that runs the app later on... so nevermind my comment. 

But some other things just came up my mind: 
- If I understand it right, we link the java + javac version depending on which script I use and it will be build for that version due to that. But what if someone specifically uses in his `build.gradle` something like:

```
apply plugin: 'java'

compileJava {
    sourceCompatibility = 17   // not sure about right syntax :)
    targetCompatibility = 17
}
```
but picks use-j11.sh ... do we just treat this as a user misconfiguration?

- Similar to the previous one, since gradle 6.7 you can do
```
java {
    toolchain {
        languageVersion.set(JavaLanguageVersion.of(11))
    }
}
```
which does according to the [doc](https://docs.gradle.org/current/userguide/toolchains.html#sec:consuming):

> 1. Setup all compile, test and javadoc tasks to use the defined toolchain which may be different than the one Gradle itself uses
> 2. Gradle detects locally installed JVMs
> 3. Gradle chooses a JRE/JDK matching the requirements of the build (in this case a JVM supporting Java 11)
> 4. If no matching JVM is found, it will automatically download a matching JDK from AdoptOpenJDK

which is also another possibility to determine the target java version next to the use-jxx.sh scripts. Do you see that as an issue? 

Btw. What I like about this toolchain thing is that you could potentially do a build with every java version you like to independently of the jenkins-agents version (which would solve the general issue we have in this thread)... but on the downside this would download the adoptopenjdk with every pipeline run in case it differs from the one installed on the agent. I guess the downside is worse here, what do you think? Also this does not help if people use maven over gradle.

Reply:
I had no problem in offering and maintaining two agents in ODS. I'm proposing a solution for one agent because Michael raise the maintenance cost as issue. Anyway at the moment the `use-j17.sh` provide a simple straightforward solution which I believe is easy to maintain and is not complex at all.

 > but picks use-j11.sh ... do we just treat this as a user misconfiguration? The issue is that we now have more than one place to determine the java version it should be built with.

gradle will fail in this case. This PR upgrades the spring boot quickstarter to use `java 17` and adds `use-j17.sh` in the Jenkinsfile to switch the java version. The default will be java 11, so that backwards compatibility is guaranteed for project upgrading to ODS 5.x

> 2. Similar to the previous one, since gradle 6.7 you can do...

I don't this as an issue at the moment. The problem of downloading the jdk on the fly is the performance penalty on the build duration.

@renedupont do you see any blocker, or are you fine with this PR? I'd like to convert from draft to a real PR soon.